### PR TITLE
Add template matcher for SMS messages

### PR DESCRIPTION
**Pull request by adrianclay**
We have a number of GOV.UK Notify templates for the SMS messages
depending on what message you send through to the service.

This behaviour of what you get sent has been matched to that of
the PHP app.

### CODE REVIEW

File: lib/sms_template_finder.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class SmsTemplateFinder
+  def execute(message_content:, env:)
+    device_name_matchers.each do |matcher, device_name|
+      return device_instruction_config(env).fetch(device_name) if message_content.match?(matcher)
+    end
+
+    template_name = 'generic_help'
+    template_name = 'credentials' if message_content == '' || message_content.match(/go/i)
+    template_name = 'help_menu' if message_content == 'help'
```
Comment (adrianclay) [inline lib/sms_template_finder.rb]: Do we want to make this a looser matcher?

File: lib/sms_template_finder.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class SmsTemplateFinder
+  def execute(message_content:, env:)
+    device_name_matchers.each do |matcher, device_name|
+      return device_instruction_config(env).fetch(device_name) if message_content.match?(matcher)
+    end
+
+    template_name = 'generic_help'
+    template_name = 'credentials' if message_content == '' || message_content.match(/go/i)
```
Comment (adrianclay) [inline lib/sms_template_finder.rb]: Do we want to match if you send through a blank space?

File: lib/sms_template_finder.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+private
+
+  def device_instruction_config(env)
+    config(env).fetch('device_help')
+  end
+
+  def device_name_matchers
+    {
+      /1|android|samsung|galaxy|htc|huawei|sony|motorola|lg|nexus/i => 'android',
```
Comment (emileswarts) [inline lib/sms_template_finder.rb]: Would be nice to see all these regexes grouped and assigned to constants with descriptive names.

File: spec/lib/sms_template_finder_spec.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      end
+    end
+  end
+
+
+  context 'Given there is a Go message' do
+    it 'returns the user details template id' do
+      expect(template_for_message('go')).to eq('3a4b1ca8-7b26-4266-8b5f-e05fdbd11879')
+      expect(template_for_message('Go')).to eq('3a4b1ca8-7b26-4266-8b5f-e05fdbd11879')
+      expect(template_for_message('GO')).to eq('3a4b1ca8-7b26-4266-8b5f-e05fdbd11879')
```
Comment (emileswarts) [inline spec/lib/sms_template_finder_spec.rb]: Could refactor out this duplication with a let, ok with this for now.

**Pull request merged**